src/bootstrap/src/core/build_steps/setup.rs

```diff
@@ -1,3 +1,9 @@
+//! First time setup of a dev environment
+//!
+//! These are build-and-run steps for `./x.py setup`, which allows quickly setting up the directory
+//! for modifying, building, and running the compiler and library. The main convenience is to allow
+//! not having to painstakingly set every single option in config.toml.
```

Comment:
oh, and originally one of the goals was to avoid the trap where someone runs x.py build before configuring a profile and now they have the LLVM submodule checked out indefinitely and their life is pain. for a while now LLVM is downloaded by default if you don't have a config.toml, though, so that doesn't really apply anymore.

Reply:
> it makes profiles more discoverable; most people's instinct is to configure options one by one and that's a waste of their time.

okay but isn't that just saying what I said with different words tho' <_<;